src/main/java/seedu/address/logic/parser/PayCommandParser.java

```diff
@@ -0,0 +1,34 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.PayCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Money;
+
+/**
+ * Parses input arguments and creates a new PayCommand object
+ */
+public class PayCommandParser implements Parser<PayCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the PayCommand
+     * and returns a PayCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public PayCommand parse(String args) throws ParseException {
+        try {
+            Index index = ParserUtil.parseIndex(args.trim().split("\\s+", 2)[0]);
+            Money moneyPaid = ParserUtil.parseMoney(args.trim().split("\\s+", 2)[1]);
+            return new PayCommand(index, moneyPaid);
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, PayCommand.MESSAGE_USAGE), pe);
+        } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
I think whether `Index` is out of range of `Array` is never checked in this block of code and hence there is no point to have such a `catch block` here?

Reply:
In order to use the current implementation of parseIndex() and parseMoney(), I have to split the user's command first using white spaces. If the user did not type the command correctly, for example `pay 2 {missing money}`, my split function will throw an out of bounds error. This is to catch that exception

Reply:
Ahh I see, make sense to me!